heron/stmgr/src/cpp/manager/stmgr-clientmgr.cpp

```diff
@@ -141,6 +141,14 @@ bool StMgrClientMgr::SendTupleStreamMessage(sp_int32 _task_id, const sp_string&
   out = __global_protobuf_pool_acquire__(out);
   out->set_task_id(_task_id);
   out->set_src_task_id(_msg.src_task_id());
+  sp_int32 length = 0;
+  if (_msg.has_data()) {
```

Comment:
The `length` here represents both data tuple size and control tuple size. The mixing might cause confusions.

Reply:
An alternate design would be to have separate optional length fields for control and data. I can do that or I can use the current field for data only.